src/v2/view-builder/views/idp/AuthenticatorIdPView.js

```diff
@@ -0,0 +1,25 @@
+import EnrollIdPForm from './EnrollAuthenticatorIdPForm';
+import ChallengeIdpForm from './ChallengeAuthenticatorIdPForm';
+import BaseAuthenticatorView from '../../components/BaseAuthenticatorView';
+import AuthenticatorEnrollFooter from '../../components/AuthenticatorEnrollFooter';
+import AuthenticatorVerifyFooter from '../../components/AuthenticatorVerifyFooter';
+
+export default BaseAuthenticatorView.extend({
+  initialize () {
+    BaseAuthenticatorView.prototype.initialize.apply(this, arguments);
+
+    const currentAuthenticator = this.options.appState.get('currentAuthenticator');
+    const authenticatorEnrollments = this.options.appState.get('authenticatorEnrollments')?.value || [];
+    const existingAuthenticator = authenticatorEnrollments
+      .filter(authenticator => authenticator.id === currentAuthenticator.id);
+
+    // Set the view based on whether or not we're in a challenge flow
+    this.Body = existingAuthenticator.length
```

Comment:
this could be 
```
let options = existingAuthenticator.length ? options1 : options2;
this.Body = new IdpForm({options}) // we can avoid duplicate forms
```